pkg/engine/internal/executor/expressions.go

```diff
@@ -148,6 +153,22 @@ func (e expressionEvaluator) eval(expr physical.Expression, input arrow.Record)
 			return nil, fmt.Errorf("failed to lookup binary function for signature %v(%v,%v): %w", expr.Op, lhs.Type(), rhs.Type(), err)
 		}
 		return fn.Evaluate(lhs, rhs)
+
+	case *physical.FunctionExpr:
+		args := make([]ColumnVector, len(expr.Expressions))
+		for i, arg := range expr.Expressions {
+			p, err := e.eval(arg, input)
+			if err != nil {
+				return nil, err
+			}
+			args[i] = p
+		}
```

Comment:
I think we need to find a way to optimize this at some point.
Parsing the argument expressions every time for each batch is a lot of overhead, especially also because these are always string literals (aren't they) and therefore have a single value across all rows.

Reply:
At least for now. Later, when we support `| logfmt foo,bar` this may become a problem.